src/components/image/image.tsx

```diff
@@ -81,21 +103,32 @@ export class EuiImage extends Component<EuiImageProps, State> {
       allowFullScreen,
       fullScreenIconColor = 'light',
       alt,
+      style,
       ...rest
     } = this.props;
 
     const { isFullScreenActive } = this.state;
+    const customStyle: React.CSSProperties = { ...style };
 
-    const classes = classNames(
+    let classes = classNames(
       'euiImage',
-      sizeToClassNameMap[size],
       {
         'euiImage--hasShadow': hasShadow,
         'euiImage--allowFullScreen': allowFullScreen,
       },
       className
     );
 
+    if (typeof size === 'string' && SIZES.includes(size)) {
+      classes = `${classes} ${sizeToClassNameMap[size as ImageSize]}`;
+    } else {
+      classes = `${classes}`;
+      customStyle.maxWidth = size;
+      customStyle.maxHeight = size;
+      customStyle.height = 'auto';
+      customStyle.width = 'auto';
```

Comment:
From what I can tell, only width=auto is needed since your overriding the CSS `width = 100%`. Also, lets add a comment so we don't have to keep describing the necessity of this

```suggestion
      // Set width back to auto to ensure aspect ratio is kept
      customStyle.width = 'auto';
```